### PR TITLE
EXLM-18 Browse All Left Nav block

### PR DESCRIPTION
**Pull request by ruhisingh1**
Please always provide the Jira Issue your PR is for, as well as test URLs where your change can be observed (before and after):

Jira ID: EXLM-18

Test URLs:

- Before: https://main--exlm--adobe-experience-league.hlx.page/en/browse
- After: https://feature-exlm-18--exlm--adobe-experience-league.hlx.live/en/browse
            https://feature-exlm-18--exlm--adobe-experience-league.hlx.live/en/browse/experience-manager/experience-manager-6-5/experience-manager-sites

### CODE REVIEW

**Review by Nehaadobe — APPROVED** [review]

File: styles/styles.css [hunk 1]
```diff
@@ -751,7 +751,7 @@ img {
   padding: 30px 16px;
   /* section metadata */
 }
-.section[data-section-status=loading], .section[data-section-status=initialized]{
+.section[data-section-status=loading], .section[data-section-status=initialized] {
```
Comment (noormd08) [inline styles/styles.css:754]: This seems to be very generic. Hope it is not breaking other sections. It is better to have the CSS specific to that block name and keep it here.

File: styles/styles.scss [hunk 2]
```diff
@@ -681,6 +682,7 @@ main > div.browse-rail {
     /* on pages with rails the max width must be on main */
     max-width: var(--non-spectrum-max-width);
     margin: auto;
+    grid-template-rows: auto auto;
```
Comment (noormd08) [inline styles/styles.scss:685]: You can remove second value from here.

File: blocks/browse-rail/browse-rail.js [hunk 3]
```diff
@@ -1,9 +1,238 @@
...
+function toggleItemVisibility(itemList, startIndex, show) {
+  // eslint-disable-next-line no-plusplus
+  for (let i = startIndex; i < itemList.length; i++) {
+    itemList[i].classList.toggle('hidden', !show);
+  }
+}
+
+// Utility function to set link visibility
+function setLinkVisibility(linkClass, show) {
+  const linkElement = document.querySelector(linkClass);
```
Comment (noormd08) [inline blocks/browse-rail/browse-rail.js:15]: Can't this be done based on block?

File: blocks/browse-rail/browse-rail.js [hunk 4]
```diff
@@ -1,9 +1,238 @@
...
+// Utility Function to get the page path until a specific level
+function getPathUntilLevel(originalUrl, levels) {
+  const pathSegments = originalUrl.split('/');
+  const resultPath = pathSegments.slice(0, levels + 1).join('/');
+  return resultPath;
+}
+
+// Function to handle "View More" click
+function handleViewMoreClick() {
+  const itemList = document.querySelectorAll('.products > li > ul > li');
```
Comment (noormd08) [inline blocks/browse-rail/browse-rail.js:42]: Same as above

File: blocks/browse-rail/browse-rail.js [hunk 5]
```diff
@@ -1,9 +1,238 @@
...
+  toggleItemVisibility(itemList, 12, true);
+  setLinkVisibility('.viewMoreLink', false);
+  setLinkVisibility('.viewLessLink', true);
+}
+
+// Function to handle "View Less" click
+function handleViewLessClick() {
+  const itemList = document.querySelectorAll('.products > li > ul > li');
```
Comment (noormd08) [inline blocks/browse-rail/browse-rail.js:50]: Same as above

File: blocks/browse-rail/browse-rail.js (hunk 4 above)
Comment (ruhisingh1) [reply blocks/browse-rail/browse-rail.js]: Done

File: blocks/browse-rail/browse-rail.js (hunk 5 above)
Comment (ruhisingh1) [reply blocks/browse-rail/browse-rail.js]: Done

File: styles/styles.css [hunk 6]
```diff
@@ -792,6 +792,7 @@ main > div.browse-rail {
     /* on pages with rails the max width must be on main */
     max-width: var(--non-spectrum-max-width);
     margin: auto;
+    grid-template-rows: auto auto;
```
Comment (Nehaadobe) [inline styles/styles.css:795]: @noormd08 , this is used to avoid extra space for rows in the grid. Single value won't work.

**Review by Nehaadobe — APPROVED** [review]

**Review by noormd08 — APPROVED** [review]

**Pull request merged**